goobi_adapter/goobi_reader/src/test/scala/uk/ac/wellcome/platform/goobi_reader/services/GoobiReaderWorkerServiceTest.scala

```diff
@@ -0,0 +1,321 @@
+package uk.ac.wellcome.platform.goobi_reader.services
+
+import java.io.{ByteArrayInputStream, InputStream}
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+
+import akka.actor.ActorSystem
+import com.amazonaws.services.s3.{AmazonS3, AmazonS3Client}
+import org.mockito.Matchers.any
+import org.mockito.Mockito.when
+import org.scalatest.FunSpec
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.mockito.MockitoSugar
+import uk.ac.wellcome.exceptions.GracefulFailureException
+import uk.ac.wellcome.messaging.sns.NotificationMessage
+import uk.ac.wellcome.messaging.sqs.{SQSConfig, SQSStream}
+import uk.ac.wellcome.messaging.test.fixtures.SQS.Queue
+import uk.ac.wellcome.monitoring.MetricsSender
+import uk.ac.wellcome.monitoring.test.fixtures.MetricsSenderFixture
+import uk.ac.wellcome.platform.goobi_reader.GoobiRecordMetadata
+import uk.ac.wellcome.platform.goobi_reader.fixtures.GoobiReaderFixtures
+import uk.ac.wellcome.storage.dynamo._
+import uk.ac.wellcome.storage.s3.S3StreamStore
+import uk.ac.wellcome.storage.test.fixtures.LocalDynamoDb.Table
+import uk.ac.wellcome.storage.test.fixtures.S3.Bucket
+import uk.ac.wellcome.storage.vhs.{HybridRecord, VersionedHybridStore}
+import uk.ac.wellcome.test.fixtures.{fixture, Akka, TestWith}
+import uk.ac.wellcome.test.utils.ExtendedPatience
+
+import scala.concurrent.duration._
+
+class GoobiReaderWorkerServiceTest
+    extends FunSpec
+    with Akka
+    with MetricsSenderFixture
+    with ScalaFutures
+    with ExtendedPatience
+    with MockitoSugar
+    with GoobiReaderFixtures {
+
+  private val id = "mets-0001"
+  private val goobiS3Prefix = "goobi"
+  private val sourceKey = s"$id.xml"
+  private val contents = "muddling the machinations of morose METS"
+  private val updatedContents = "Updated contents"
+  private val eventTime = Instant.parse("2018-01-01T01:00:00.000Z")
+  private val contentsStorageKey = s"$goobiS3Prefix/10/$id/cd92f8d3"
+  private val updatedContentsStorageKey = s"$goobiS3Prefix/10/$id/536b51f7"
+
+  it("processes a notification") {
+    withGoobiReaderWorkerService(s3Client) {
+      case (bucket, queue, table, workerService) =>
+        s3Client.putObject(bucket.name, sourceKey, contents)
+
+        workerService.processMessage(
```

Comment:
I don't think we should test `processMessage` directly cause that would pass eve if the stream was never started in the test.